drivers/gpio/gpio_nrfx.c

```diff
@@ -55,7 +55,7 @@ static int gpiote_channel_alloc(u32_t abs_pin, nrf_gpiote_polarity_t polarity)
 			nrf_gpiote_event_configure(channel, abs_pin, polarity);
 			nrf_gpiote_event_enable(channel);
 			nrf_gpiote_int_enable(BIT(channel));
-			return channel;
```

Comment:
Why is this an issue if one checks for return value `>=0` ?

Reply:
Return from `gpiote_pin_int_cfg` is used directly by `gpio_pin_enable_callback` and `gpio_pin_disable_callback`.

Brief from `gpio_pin_enable_callback` API:

>>* @return 0 if successful, negative errno code on failure.

Application was not expecting a positive value from this call. I can fix the application but the problem is that in many places code checks are `if (err)`. It would be huge problem if underlying layer start to return the positive values without mentioning them in the API.

Reply:
Not sure I follow completely the explanation from @pdunaj , however, I like this change. This is aligned with several similar cases in Zephyr, where 0 indicates success and non-zero errors map to error codes, reflecting error conditions.

Reply:
A follow-up question; how does the upper layers get to know the  exact channel that was allocated?

Reply:
@ioannisg does it need to know? From the point of view of the application we are enabling/disabling callback on a pin. How channel refers to it?
If it is important for any reason maybe there should be a specific API created?

Reply:
Np, I  was just wondering how and if the channel info is needed by the  upper layer.

Reply:
It is not needed now, it can stay this way.